### ISSUE TITLE
Consumption to EP Migration - via CLI

### ISSUE THREAD

**Issue by cachai2** [Azure/azure-cli#15252]
Create an experience to allow Azure Functions users to upgrade from a consumption plan to a premium plan through the CLI. With the new command, we will need to ask the user for sku and size for target, and all additional information when creating the premium plan can be inferred from the source plan.

**Comment by yonzhan** [comment]
Functions

**Comment by fabiocav** [comment]
@gzuber for awareness

**Comment by gzuber** [comment]
While discussing the specifics of where this command should be implemented, we found the command az functionapp update --plan which allows a user to swap a functionapp between app service plans. This effectively accomplishes this upgrade scenario to a point where making another command might be redundant. The existing command doesn't create a new plan for the customer, so it's not exactly the same but very close.

For now, we've decided to stick with this old command. However, this command enables a user to switch from an ElasticPremium plan to a consumption plan which is potentially dangerous. Created Azure/azure-cli#15694 to make changes to adequately inform a user of the risks involved.
